packages/ag-charts-community/src/chart/fonts/fontManager.ts

```diff
@@ -0,0 +1,52 @@
+import type { DOMManager } from '../../dom/domManager';
+import { ChartUpdateType } from '../chartUpdateType';
+import type { UpdateService } from '../updateService';
+
+export class FontManager {
+    constructor(
+        private readonly domManager: DOMManager,
+        private readonly updateService: UpdateService
+    ) {}
+
+    public updateFonts(fonts?: Set<string>) {
+        if (!fonts || fonts.size === 0) return;
+        this.loadFonts(fonts);
+        fonts.forEach((font) => {
+            this.observeFontStatus(font);
+        });
+    }
+
+    private loadFonts(fonts: Set<string>) {
+        const fontStrings = Array.from(fonts).map((font) => encodeURIComponent(font));
+        const css = `@import url('https://fonts.googleapis.com/css2?family=${fontStrings.join('&family=')}:wght@100;200;300;400;500;600;700;800;900&display=swap');\n`;
+        this.domManager.addStyles(`google-font-${fontStrings.join('-')}`, css);
+    }
+
+    private observeFontStatus(font: string) {
+        const fontCheckElement = window.document.createElement('div');
+        fontCheckElement.style.setProperty('position', 'absolute');
+        fontCheckElement.style.setProperty('top', '0');
+        fontCheckElement.style.setProperty('margin', '0');
+        fontCheckElement.style.setProperty('padding', '0');
+        fontCheckElement.style.setProperty('overflow', 'hidden');
+        fontCheckElement.style.setProperty('visibility', 'hidden');
+        fontCheckElement.style.setProperty('width', 'auto');
+        fontCheckElement.style.setProperty('max-width', 'none');
+        fontCheckElement.style.setProperty('font-synthesis', 'none');
+        fontCheckElement.style.setProperty('font-family', font);
+        fontCheckElement.style.setProperty('font-size', '16px');
+        fontCheckElement.style.setProperty('white-space', 'nowrap');
+        fontCheckElement.textContent = 'UVWxyz';
+
+        this.domManager.addChild('canvas-container', `font-check-${encodeURIComponent(font)}`, fontCheckElement);
+
+        // Observe changes to the element size as a proxy for the font loading
+        const fontCheckObserver = new ResizeObserver((entries) => {
+            const width = entries?.at(0)?.contentBoxSize.at(0)?.inlineSize;
+            if (width != null && width > 0) {
+                this.updateService.update(ChartUpdateType.PERFORM_LAYOUT);
+            }
+        });
+        fontCheckObserver.observe(fontCheckElement);
```

Comment:
The observer is not destroyed because there may be multiple fonts in the string, some of which may be available locally and others need to be loaded. So the observer will trigger multiple times.

I'll do some more extensive testing to see if this can be checked and destroyed after either a known number of times or (not preferable) a timeout.

Reply:
Could we at least ensure that things are completed cleaned up on `chart.destroy()` to avoid a memory leak?

Reply:
Added `destroy()` for the observers, elements are destroyed by the `DOMManager`.